skopt/optimizer/optimizer.py

```diff
@@ -176,7 +189,74 @@ def _check_arguments(self, base_estimator, n_random_starts, acq_optimizer):
                              "'sampling', got {0}".format(acq_optimizer))
         self.acq_optimizer = acq_optimizer
 
-    def ask(self):
+        par_strategies = ["cl_min", "cl_mean", "cl_max"]
+        if not parallel_strategy in par_strategies:
+            raise ValueError(
+                "Expected parallel_strategy to be one of " + str(par_strategies) + ", "
+                "got %s" % parallel_strategy)
+        self.parallel_strategy = parallel_strategy
+
+    def copy(self):
+        """Creates a shallow copy of optimizer."""
+        optimizer = Optimizer(
+            dimensions=self.space.dimensions,
+            base_estimator=self.base_estimator,
+            n_random_starts=self._n_random_starts,
+            acq_func=self.acq_func,
+            acq_optimizer=self.acq_optimizer,
+            random_state=copy.copy(self.rng),
+            acq_func_kwargs=self.acq_func_kwargs,
+            acq_optimizer_kwargs=self.acq_optimizer_kwargs,
+            parallel_strategy=self.parallel_strategy
+        )
+
+        if hasattr(self, "gains_"):
+            optimizer.gains_ = np.copy(self.gains_)
+
+        if len(self.Xi) > 0:
+            optimizer.tell(self.Xi, self.yi)
+
+        return optimizer
+
+    def ask(self, n_points=None):
+        """Query point or multiple points to evaluate.
+
+        * `n_jobs` [int or None]:
+            Number of function evaluations that can be done in parallel.
+            If the value is None, a single point to evaluate is returned.
+            Otherwise a list of points to evaluate is returned of size n_jobs.
+        """
+        if n_points is None:
+            return self._ask()
+
+        opt = self.copy()
+
+        # when _n_random_starts > 0, random state does not change
+        # during tell(), but only during ask(). As ask() is called
+        # on a copy of the optimizer only, the random state of original
+        # optimizer does not change during _n_random_starts > 0. If
+        # the random state is not shared with original optimizer,
+        # then same points are generated, which leads to exceptions
+        # with fitting of some models.
+        if self._n_random_starts > 0:
```

Comment:
I think I understood the problem, but can you give an example just for some clarity. Do you mean that if this line is removed?

```python
opt = Optimizer(n_random_starts=2)
Xi = opt.ask(n_points=2)
opt.tell(Xi, yi)
x = opt.ask()
opt.tell(Xi + x, yi + yi[0])
```

will fail because Xi[0] is the same as x?